artemis-server/src/main/java/org/apache/activemq/artemis/core/management/impl/view/ProducerView.java

```diff
@@ -79,7 +79,7 @@ public Object getField(ServerProducer producer, String fieldName) {
          case "user":
             return session.getUsername();
          case "clientID":
-            return session.getRemotingConnection().getClientID();
+            return session.getMetaData("jms-client-id") == null ? session.getRemotingConnection().getClientID() : session.getMetaData("jms-client-id");
```

Comment:
this should not be needed, if anything the session and producer models should be re-worked to allow for a client id to be set at session level generically.